common_features.mk

```
@@ -82,7 +82,7 @@ ifeq ($(strip $(AUDIO_ENABLE)), yes)
     SRC += $(QUANTUM_DIR)/process_keycode/process_audio.c
     SRC += $(QUANTUM_DIR)/process_keycode/process_clicky.c
     SRC += $(QUANTUM_DIR)/audio/audio.c ## common audio code, hardware agnostic
-    SRC += $(QUANTUM_DIR)/audio/driver_$(PLATFORM_KEY)_$(strip $(AUDIO_DRIVER)).c
+    SRC += audio_$(strip $(AUDIO_DRIVER)).c
```

```suggestion
    SRC += $(PLATFORM_PATH)/$(PLATFORM_KEY)/$(DRIVER_DIR)/audio_$(strip $(AUDIO_DRIVER)).c
```
Without this I think the .o files will be placed in the root of the build directory. It'll still work, but it's nice to keep the same folder structure as the source.

---

You are correct, and yeah, it may be better to do that.

But I don't think the chance of collision is very high, considering that it does append the driver type to the name, and audio.c is in the quantum folder.  But I'm open to suggestions.

---

Yeah, it's not so much about collision as it is about having the build folder hierarchy match the source folder hierarchy.